cohortextractor/query_language.py

```diff
@@ -20,20 +23,29 @@ class Comparator:
     """A generic comparator to represent a comparison between a source object and a value"""
 
     def __init__(
-        self, children=None, connector="and_", source=None, operator=None, value=None
+        self,
+        children=None,
```

Comment:
I think I'm slightly confused about the role of `children` vs `lhs` and `rhs` here. (I know this isn't something you're adding in this PR and so it's not something I'm suggesting changing now, I just haven't quite got my head around it.)

For an operation like `__and__` we add the two nodes to the `children` list, but why can't they just be recorded as `lhs` and `rhs` attributes like others do?

Reply:
Maybe they can?  I may have been misled by using django's `Q` as a model, and thinking that there could be more than 2 children at some point.  I'll give it a try (in a separate PR)